==3822== Memcheck, a memory error detector
==3822== Copyright (C) 2002-2011, and GNU GPL'd, by Julian Seward et al.
==3822== Using Valgrind-3.7.0 and LibVEX; rerun with -h for copyright info
==3822== Command: TestCollatz
==3822== 
Running main() from gtest_main.cc
[==========] Running 19 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 19 tests from Collatz
[ RUN      ] Collatz.read
[       OK ] Collatz.read (30 ms)
[ RUN      ] Collatz.readA
[       OK ] Collatz.readA (5 ms)
[ RUN      ] Collatz.readB
[       OK ] Collatz.readB (4 ms)
[ RUN      ] Collatz.readC
[       OK ] Collatz.readC (4 ms)
[ RUN      ] Collatz.eval_1
[       OK ] Collatz.eval_1 (6 ms)
[ RUN      ] Collatz.eval_2
[       OK ] Collatz.eval_2 (3 ms)
[ RUN      ] Collatz.eval_3
[       OK ] Collatz.eval_3 (3 ms)
[ RUN      ] Collatz.eval_4
[       OK ] Collatz.eval_4 (3 ms)
[ RUN      ] Collatz.eval_5
[       OK ] Collatz.eval_5 (4 ms)
[ RUN      ] Collatz.eval_6
[       OK ] Collatz.eval_6 (3 ms)
[ RUN      ] Collatz.eval_7
==3822== Invalid read of size 4
==3822==    at 0x404C47: collatz_eval(int, int) (in /v/filer4b/v38q001/alsabag/OOP/collatz/cs371p-collatz./collatz/TestCollatz)
==3822==    by 0x40800D: Collatz_eval_7_Test::TestBody() (in /v/filer4b/v38q001/alsabag/OOP/collatz/cs371p-collatz./collatz/TestCollatz)
==3822==    by 0x42D1BC: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/alsabag/OOP/collatz/cs371p-collatz./collatz/TestCollatz)
==3822==    by 0x425816: testing::Test::Run() (in /v/filer4b/v38q001/alsabag/OOP/collatz/cs371p-collatz./collatz/TestCollatz)
==3822==    by 0x4258EC: testing::TestInfo::Run() (in /v/filer4b/v38q001/alsabag/OOP/collatz/cs371p-collatz./collatz/TestCollatz)
==3822==    by 0x425A36: testing::TestCase::Run() (in /v/filer4b/v38q001/alsabag/OOP/collatz/cs371p-collatz./collatz/TestCollatz)
==3822==    by 0x425CC6: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/alsabag/OOP/collatz/cs371p-collatz./collatz/TestCollatz)
==3822==    by 0x42CD3C: bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (in /v/filer4b/v38q001/alsabag/OOP/collatz/cs371p-collatz./collatz/TestCollatz)
==3822==    by 0x424EBA: testing::UnitTest::Run() (in /v/filer4b/v38q001/alsabag/OOP/collatz/cs371p-collatz./collatz/TestCollatz)
==3822==    by 0x404824: main (in /v/filer4b/v38q001/alsabag/OOP/collatz/cs371p-collatz./collatz/TestCollatz)
==3822==  Address 0xc00690 is not stack'd, malloc'd or (recently) free'd
==3822== 
==3822== 
==3822== Process terminating with default action of signal 11 (SIGSEGV)
==3822==  Access not within mapped region at address 0xC00690
==3822==    at 0x404C47: collatz_eval(int, int) (in /v/filer4b/v38q001/alsabag/OOP/collatz/cs371p-collatz./collatz/TestCollatz)
==3822==    by 0x40800D: Collatz_eval_7_Test::TestBody() (in /v/filer4b/v38q001/alsabag/OOP/collatz/cs371p-collatz./collatz/TestCollatz)
==3822==    by 0x42D1BC: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/alsabag/OOP/collatz/cs371p-collatz./collatz/TestCollatz)
==3822==    by 0x425816: testing::Test::Run() (in /v/filer4b/v38q001/alsabag/OOP/collatz/cs371p-collatz./collatz/TestCollatz)
==3822==    by 0x4258EC: testing::TestInfo::Run() (in /v/filer4b/v38q001/alsabag/OOP/collatz/cs371p-collatz./collatz/TestCollatz)
==3822==    by 0x425A36: testing::TestCase::Run() (in /v/filer4b/v38q001/alsabag/OOP/collatz/cs371p-collatz./collatz/TestCollatz)
==3822==    by 0x425CC6: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/alsabag/OOP/collatz/cs371p-collatz./collatz/TestCollatz)
==3822==    by 0x42CD3C: bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (in /v/filer4b/v38q001/alsabag/OOP/collatz/cs371p-collatz./collatz/TestCollatz)
==3822==    by 0x424EBA: testing::UnitTest::Run() (in /v/filer4b/v38q001/alsabag/OOP/collatz/cs371p-collatz./collatz/TestCollatz)
==3822==    by 0x404824: main (in /v/filer4b/v38q001/alsabag/OOP/collatz/cs371p-collatz./collatz/TestCollatz)
==3822==  If you believe this happened as a result of a stack
==3822==  overflow in your program's main thread (unlikely but
==3822==  possible), you can try to increase the size of the
==3822==  main thread stack using the --main-stacksize= flag.
==3822==  The main thread stack size used in this run was 8388608.
==3822== 
==3822== HEAP SUMMARY:
==3822==     in use at exit: 5,983 bytes in 107 blocks
==3822==   total heap usage: 703 allocs, 596 frees, 74,356 bytes allocated
==3822== 
==3822== LEAK SUMMARY:
==3822==    definitely lost: 0 bytes in 0 blocks
==3822==    indirectly lost: 0 bytes in 0 blocks
==3822==      possibly lost: 1,190 bytes in 38 blocks
==3822==    still reachable: 4,793 bytes in 69 blocks
==3822==         suppressed: 0 bytes in 0 blocks
==3822== Rerun with --leak-check=full to see details of leaked memory
==3822== 
==3822== For counts of detected and suppressed errors, rerun with: -v
==3822== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 2 from 2)
